datafusion/expr/src/expr_rewriter.rs

```diff
@@ -429,27 +433,14 @@ pub fn normalize_col_with_schemas(
     schemas: &[&Arc<DFSchema>],
     using_columns: &[HashSet<Column>],
 ) -> Result<Expr> {
-    struct ColumnNormalizer<'a> {
```

Comment:
Here is a pretty good example the boilerplate that can be avoided by using `rewrite_expr` rather than `ExprRewriter` (rust generates a closure that captures `schemas` and `using_columns`) rather than having to do it explicitly

Reply:
Really like this example.👍